graphql/authorization/auth.go

```diff
@@ -39,33 +43,56 @@ var (
 )
 
 type AuthMeta struct {
-	PublicKey string
-	Header    string
-	Namespace string
+	HMACPublicKey string
+	RSAPublicKey  *rsa.PublicKey
+	Header        string
+	Namespace     string
+	Algo          string
 }
 
-func (m *AuthMeta) Parse(schema string) {
+func (m *AuthMeta) Parse(schema string) error {
 	poundIdx := strings.LastIndex(schema, "#")
 	if poundIdx == -1 {
-		return
+		return nil
 	}
 	lastComment := schema[poundIdx:]
-	length := strings.Index(lastComment, "\n")
-	if length > -1 {
-		lastComment = lastComment[:length]
+	if !strings.HasPrefix(lastComment, "# Authorization") {
+		return nil
+	}
+
+	authMetaRegex, err :=
+		regexp.Compile(`^#[\s]([^\s]+)[\s]+([^\s]+)[\s]+([^\s]+)[\s]+([^\s]+)[\s]+"([^\"]+)"`)
+	if err != nil {
+		return errors.Errorf("error while parsing jwt authorization info: %v", err)
+	}
+	idx := authMetaRegex.FindAllStringSubmatchIndex(lastComment, -1)
+	if len(idx) != 1 || len(idx[0]) != 12 ||
+		!strings.HasPrefix(lastComment, lastComment[idx[0][0]:idx[0][1]]) {
```

Comment:
Could you add a comment of the format of the header etc maybe at the function Parse() ccomments ? It is hard to follow without it.